@uppy/provider-views: fix ProviderView error on empty plugin.icon

dschmidt opened the pull request:
The dashboard provides a default/fallback icon for plugins without icon. The ProviderView does not - so a plugin that can be shown just fine on the overview panel, produces an error when it's actually opened. Let's use the same fallback icon.

dschmidt commented on packages/@uppy/provider-views/src/ProviderView/ProviderView.jsx:
```diff
@@ -347,11 +355,15 @@ export default class ProviderView extends View {
     const hasInput = filterInput !== ''
+    const pluginIcon = () => {
+      return this.plugin.icon?.() || defaultPickerIcon()
+    }
```
I don't really know (p)react, do I need to wrap it like this for reactivity or can I just do `const pluginIcon = this.plugin.icon || defaultPickerIcon `

arturi replied:
I think you can do the latter, yes, since that's how we pass it in the Dashboard.

dschmidt commented:
.... or I just add it to UIPlugin base class, then there's no need for any fallback logic anywhere

dschmidt commented:
> .... or I just add it to UIPlugin base class, then there's no need for any fallback logic anywhere

@arturi  What about this approach?

dschmidt commented:
Rebased and simplified the code a little.

I'm happy with this version. If you prefer this over adding the icon to the base class, let's go for this implementation

dschmidt commented:
Anything missing? Can we merge?

Murderlon commented:
Thank you!
